packages/server/api/src/app/flows/flow/flow.service.ts

```diff
@@ -287,20 +286,26 @@ export const flowService = (log: FastifyBaseLogger) => ({
                             projectId,
                             platformId,
                         })
-                        await this.updateStatus({
+                        await this.addUpdateStatusJob({
                             id,
                             projectId,
                             newStatus: operation.request.status ?? FlowStatus.ENABLED,
                         })
+                        await flowRepo().update(id, {
+                            operationStatus: operation.request.status === FlowStatus.ENABLED ? FlowOperationStatus.ENABLING : FlowOperationStatus.DISABLING,
+                        })
                         break
                     }
 
                     case FlowOperationType.CHANGE_STATUS: {
-                        await this.updateStatus({
+                        await this.addUpdateStatusJob({
                             id,
                             projectId,
                             newStatus: operation.request.status,
                         })
+                        await flowRepo().update(id, {
```

Comment:
This should happen before the adding to the queue as there could be a race condition here, ideally I would recommend this to be inside the queue job.

Same for above